### ISSUE TITLE
Sections should use h2, h3, etc.

### ISSUE THREAD

**Issue by ebruchez** [orbeon/orbeon-forms#3011]
Currently they always appear as `<h2>`. Possibly we used to have this feature a very long time ago as I see a "TODO" in the code.

[+1 from customer](https://3.basecamp.com/3600924/buckets/2016723/messages/305437441)

**Comment by ebruchez** [comment]
What would it take? Somehow, the section must know, statically or dynamically, at which level it is. A section cannot, right now, inspect its parents, because  the XBL bound element is at the root and the ancestors in the document are not exposed.

Dynamically, we might look at the bound node of a section and guess from there, but then we would need a way to generate an element by name dynamically, and we don't have that yet.

Maybe we could use `grids-and-sections.xsl` and pass the level there:

```xml
<fr:section level="3" ...>
```

Section templates are an issue: if a section template contains sub-sections, and is used more than once and at different levels, then we can't produce a level statically. Maybe for now that is an acceptable issue, as section templates rarely contain nested sections?

**Comment by ebruchez** [comment]
On the section templates, an idea would be this:

- pass a `level` parameter to the section templates too
  - easy to find because section templates are always within an `fr:section`
- on `fr:section` nested within a section template
  - place a `level` attribute starting at, say, 0
  - place a `xbl:attr="baselevel=level"` attribute
- when the `fr:section` template instantiates, it sums the values of its `baselevel` and `level` attributes

This would work because each XBL template instantiation can determine its level at that time. Using two attributes make this simpler, but there might be other solutions.
